build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/ts2EsmTask.ts

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as JSON5 from "json5";
+import { lstatSync, readFileSync } from "node:fs";
+import path from "node:path";
+import type { TsConfigJson } from "type-fest";
+import { globFn } from "../../../common/utils";
+import { LeafWithFileStatDoneFileTask } from "./leafTask";
+
+export class Ts2EsmTask extends LeafWithFileStatDoneFileTask {
+	/**
+	 * Parses the ts2esm command to determine which tsconfigs are used, then reads the input files based on the settings
+	 * in the tsconfig `includes` and `files` settings.
+	 *
+	 * @remarks
+	 *
+	 * The `excludes` property in tsconfig is NOT USED! This means the cache may look at files that it doesn't need to.
+	 */
+	protected async getInputFiles(): Promise<string[]> {
+		const inputFiles: string[] = [];
+		const split = this.command.split(" ");
+
+		// Assume arguments are package-relative paths to tsconfigs
+		const configs = split.slice(1).map((filePath) => this.getPackageFileFullPath(filePath));
+
+		for (const configPath of configs) {
+			const tsConfig = JSON5.parse(readFileSync(configPath, "utf8")) as TsConfigJson;
+			if (tsConfig.files !== undefined) {
+				inputFiles.push(...tsConfig.files);
```

Comment:
@tylerbutler, None of the code in the block common with the other tasks that read tsc config?

Reply:
No, existing tasks don't seem to parse the tsconfig directly; they hook into TypeScript stuff directly instead. I considered doing that here but it seemed more complicated than necessary.